src/main/scala/introcourse/level03/ListExercises.scala

```diff
@@ -268,5 +314,35 @@ object ListExercises {
     * Given: val l1 = List("a", "a", "a", "a", "b", "c", "c", "a", "a", "d", "e", "e", "e", "e")
     * sublists(l1) == List(List("a", "a", "a", "a"), List("b"), List("c", "c"), List("a", "a"), List("d"), List("e", "e", "e", "e"))
     */
-  def sublists[A](xs: List[A]): List[List[A]] = ???
+  def sublists[A](xs: List[A]): List[List[A]] = xs match {
+    case head :: tail => tail.foldLeft(List(List(head)))((accumulator, current) => {
+      accumulator match {
+        case _ :: _ =>
+          if (accumulator.last.contains(current)) {
+            accumulator.dropRight(1) :+ (current :: accumulator.last)
+          } else {
+            accumulator :+ List(current)
+          }
+        case Nil => Nil
+      }
+    })
+    case Nil => Nil
+  }
+
+  // [wartremover:Any] Inferred type containing Any: [_]List[_]
+  //[error]         case head :+ last => if (last.contains(current)) {
```

Comment:
Not sure why this fails sbt compile, but works when IntelliJ runs the unit tests

// TODO: 
- Find out how `sbt compile` differs from IntelliJ's compile + runtime environment
- How does scala compile and run?